docs/proposals/role-entry-points.md

```diff
@@ -0,0 +1,302 @@
+# Entry-point based roles
+
+## Variables inside a role are too open
+
+Each role consists of a set of tasks and variables read/set in some of the tasks.
+Some of the tasks can store their results into registered variables,
+e.g. store stdout of a docker pull command.
+Some of the tasks may be run only under certain conditions that are generated
```

Comment:
As @michaelgugino says. I can rephrase the "conditions generated outside of a role" to something like "conditions a role has no control over since variables inside the condition are non-role variables, e.g. inventory variables, variables set in a play or generated by the `openshift_facts`".